lib/srv/authhandlers.go

```diff
@@ -262,8 +262,13 @@ func (h *AuthHandlers) CheckFileCopying(ctx *ServerContext) error {
 }
 
 // CheckPortForward checks if port forwarding is allowed for the users RoleSet.
-func (h *AuthHandlers) CheckPortForward(addr string, ctx *ServerContext) error {
-	if ok := ctx.Identity.AccessChecker.CanPortForward(); !ok {
+func (h *AuthHandlers) CheckPortForward(addr string, ctx *ServerContext, mode services.SSHPortForwardMode) error {
+	checkedMode := ctx.Identity.AccessChecker.SSHPortForwardMode()
+	if checkedMode == services.SSHPortForwardModeOn {
+		return nil
+	}
+
+	if checkedMode == services.SSHPortForwardModeOff || checkedMode != mode {
```

Comment:
nit: checkedMode sounds like the mode we are checking, made me think it was the opposite
```suggestion
	allowedMode := ctx.Identity.AccessChecker.SSHPortForwardMode()
	if allowedMode == services.SSHPortForwardModeOn {
		return nil
	}

	if allowedMode == services.SSHPortForwardModeOff || allowedMode != mode {
```